mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/navigation/NavigationLocation.java

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.services.android.telemetry.navigation;
+
+import android.location.Location;
+import android.util.JsonToken;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.TimeZone;
+
+public class NavigationLocation {
+
+  private static final String UTC_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+  private static final String ACCURACY = "horizontalAccuracy";
+  private static final String TIMESTAMP = "timestamp";
+  private static final String ALTITUDE = "altitude";
+  private static final String COURSE = "course";
+  private static final String LONGITUDE = "lng";
+  private static final String LATITUDE = "lat";
+  private static final String SPEED = "speed";
+
+  public JSONArray getSerializedJson(Location[] locations) {
+    JSONArray jsonArray = new JSONArray();
+    for (Location loc : locations) {
+      jsonArray.put(serializeToJson(loc));
+    }
+    return jsonArray;
+  }
+
+  private JSONObject serializeToJson(Location location) {
+    formatLocationTime(location);
+
+    HashMap<String, Object> map = new HashMap<>();
+    // Potentially null values
+    map.put(ALTITUDE, location.hasAltitude() ? location.getAltitude() : JsonToken.NULL);
```

Comment:
Why `JsonToken.NULL` instead of `JSONObject.NULL`?

Reply:
Because the values null, not the entire object. It doesn't really matter though, they result in the samething.

Reply:
Wondering why not keeping consistency with the rest of the code then 🤔